lesson-16/script.js

```diff
@@ -0,0 +1,242 @@
+    (createDetailsBTN(movie.imdbID, 'imdbID')))
+
+    return li
+};
+
+const createMoviesList = (list) => {
+    const moviesList = document.createElement('ul');
+    moviesList.classList.add('list');
+
+    for (let element of list) {
+        const listItem = createListItem(element);
+        moviesList.appendChild(listItem);
+    }
+
+    return moviesList
+};
+
+const displayMovies = (moviesList) => {
+    const list = createMoviesList(moviesList);
+    const main = document.querySelector('.main');
+    main.appendChild(list);
+};
+
+function displayDetails (actors, boxOffice, genre, released, runtime) {
+    const main = document.querySelector('.main');
+
+    const detailsBlock = document.createElement('div');
+    detailsBlock.classList.add('detailsBlock');
+
+    detailsBlock.append((createParagraph(`Actors: ${actors}`, 'actors')),
+    (createParagraph(`Box office: ${boxOffice}`, 'boxOffice')),
+    (createParagraph(`Genre: ${genre}`, 'genre')),
+    (createParagraph(`Released: ${released}`, 'released')),
+    (createParagraph(`Runtime: ${runtime}`, 'runtime')))
+
+    main.appendChild(detailsBlock)
+};
+
+const displayError = () => {
+    const main = document.querySelector('.main');
+
+    const displayError = document.createElement('p');
+    displayError.classList.add('movies-list_error');
+    displayError.innerText = 'Something goes wrong. Please, try again later.';
+    main.append(displayError);
+};
+
+const loadMovies = (movieKeyWord) => {
+
+    fetch(`${requestURL}${myAPIKey}&s=${movieKeyWord}`)
+    .then((response) => {
+      return response.json();
+    })
+    .then((result) => {
+        displayMovies(result.Search);
+        countPages(result.totalResults);
+    })
+    .catch(() => {
+        displayError()
+    });
+};
+
+const loadMoviesByPage = (movieKeyWord, pageNumber) => {
+
+    fetch(`${requestURL}${myAPIKey}&s=${movieKeyWord}&page=${pageNumber}`)
+    .then((response) => {
+      return response.json();
+    })
+    .then((result) => {
+        displayMovies(result.Search);
+        countPages(result.totalResults);
+    })
+    .catch(() => {
+        displayError()
+    });
+};
+
+function loadDetailes (movieID) {
+
+    fetch(`${requestURL}${myAPIKey}&i=${movieID}`)
+    .then((response) => {
+        return response.json();
+    })
+    .then((data) => {
+        displayDetails(data.Actors, data.BoxOffice, data.Genre, data.Released, data.Runtime);
+    })
+    .catch(() => {
+        displayError()
+    });
+};
+
+document.querySelector('.header_movie-search_btn').addEventListener('click', () => {
+    const enteredKeyWord = document.querySelector('.header_movie-search_input');
+    const searchedKeyWord = document.querySelector('.header_movie-keyword');
+    searchedKeyWord.innerText = `search: "${enteredKeyWord.value}"`;
+    const movieKeyWord = enteredKeyWord.value;
+
+    clearList();
+    // clearSearch();
+    loadMovies(movieKeyWord);
+});
+
+document.addEventListener('click', (event) => {
```

Comment:
😵‍💫 

listener for whole document object?

Reply:
Better to add it for at least `main`

```suggestion
main.addEventListener('click', (event) => {
```